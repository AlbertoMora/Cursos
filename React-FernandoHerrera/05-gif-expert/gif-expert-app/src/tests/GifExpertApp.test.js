import React from "react";
import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe("Expert App tests", () => {
    test("should match component snapshot", () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });
    test("should display a list of categories", () => {
        const categories = ["Dragon Ball", "Genshin Impact"];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("GifGrid").length).toBe(categories.length);
    });
});
