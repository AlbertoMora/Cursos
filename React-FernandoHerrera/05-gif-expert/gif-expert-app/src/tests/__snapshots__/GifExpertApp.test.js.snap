// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expert App tests should display a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
    <GifGrid
      category="Genshin Impact"
      key="Genshin Impact"
    />
  </ol>
</Fragment>
`;

exports[`Expert App tests should match component snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
